File: python/48a_1692B_AllDistinct.py
def solve(arr, n):
    cnt = 0
    hset = set()
    for i in arr:
        if i in hset:
            cnt += 1
        else:
            hset.add(i)
    if cnt % 2 == 0:
        return len(hset)
    else:
        return len(hset) - 1

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = input().split()
        ans.append(solve(arr, n))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/48a_1692B_AllDistinct.cpp
#include<bits/stdc++.h>

int solve(int n){
    int cnt = 0, temp;
    std::unordered_set<int> hset;
    for(int i = 0; i < n; i++){
        std::cin>>temp;
        if(hset.find(temp) != hset.end()){
            cnt++;
        }
        else{
            hset.insert(temp);
        }
    }
    if (cnt % 2 == 0){
        return hset.size();
    }
    return hset.size() - 1;
}

int main(){
    int t, n;
    std::vector<int> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>n;
        ans.push_back(solve(n));
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}